init-distro-rid.sh

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env bash
+
+# initNonPortableDistroRid
+#
+# Input:
+#   buildOs: (str)
+#   buildArch: (str)
+#   isPortable: (int)
+#   rootfsDir: (str)
+#   isCrossBuild? (nullable:int)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# initNonPortableDistroRid will attempt to initialize a non portable rid. These
+# rids are specific to distros need to build the product/package and consume
+# them on the same platform.
+#
+# Below is the list of current non-portable platforms.
+#
+# Builds from the following *must* be non-portable:
+#
+#   |    OS     |           Expected RID            |
+#   -------------------------------------------------
+#   |   rhel6   |           rhel.6-x64              |
+#   |  alpine*  |        linux-musl-(arch)          |
+#   |  freeBSD  |        freebsd.(version)-x64      |
+#
+# It is important to note that the function does not return, but it will set
+# __DistroRid if there is a non-portable distro rid to be used.
+#
+initNonPortableDistroRid()
+{
+    # Make sure out parameter is cleared.
+    __DistroRid=
+
+    local buildOs=$1
+    local buildArch=$2
+    local isPortable=$3
+    local isCrossBuild=$4
+
+    local nonPortableBuildID=""
+    local isCrossBuild=0
+    local crossBuildLocation=""
+
+    # If there is not an optional argument passed then this will be treated
+    # as if there is no crossbuild.
+    if [ -z ${isCrossBuild} ]; then
+        crossBuildLocation=${rootfsDir}
```

Comment:
The rootfsDir is not defined anywhere now.

Reply:
Thanks this still worked because it was globally defined. Thank you for catching it either way.